### PR TITLE
Made shared_objects non-copyable

### PR DESCRIPTION
**Pull request by ice0**
Added explicit copy constructors to Transformations (fixes build errors)

### CODE REVIEW

File: synfig-core/src/modules/lyr_std/perspective.cpp [hunk 1]
```diff
@@ -352,6 +352,7 @@ namespace {
 		
 		TransformationPerspective() { }
 		explicit TransformationPerspective(const Matrix &matrix): matrix(matrix) { }
+		TransformationPerspective& operator= (const TransformationPerspective& other) { matrix = other.matrix; return *this; }
```
Comment (rodolforg) [inline synfig-core/src/modules/lyr_std/perspective.cpp:355]: what is the shared-object concept here?

**Review by rodolforg — APPROVED** [review]
Now I got it

**Comment by ice0** [comment]
@rodolforg 

Hi, sorry for the late reply.
The concept of ETL smart pointers is different from std smart pointers. 
Std smart pointers store the reference count inside the smart pointer itself. 
Unlike std, ETL stores a reference count within an object.

For this reason, you cannot put <int> object to the etl::handle. Only objects inherited from etl::shared_object.
The same concept is used in Glib::RefPtr (they switched to std::shared_ptr in gtkmm 4)
https://www.murrayc.com/permalink/2017/04/20/gtkmm-4-progress/


I removed the copy constructor just to make sure it doesn't break anything. 
But as you can see there is a code which copying `Transformations` directly, so I added explicit copy constructor to them.

**Comment by rodolforg** [comment]
@ice0 OK :) Approved then

**Pull request merged**